native/game_backend/src/lib.rs

```diff
@@ -22,17 +22,16 @@ fn parse_config(data: String) -> Config {
 fn new_game(config: Config) -> GameState {
     GameState::new(config)
 }
-
 #[rustler::nif(schedule = "DirtyCpu")]
-fn add_player(game: GameState, character_name: String) -> (GameState, Option<u64>) {
+fn add_player(game: GameState, character_name: String) -> Result<(GameState, Option<u64>), String> {
     let mut game = game;
     let player_id = game.next_id();
     match game.config.find_character(character_name) {
-        None => (game, None),
+        None => Err("Character doesn't exists".to_string()),
```

Comment:
The returned error reason should be an atom, an enum in Rust land probably

Reply:
Sounds good, do we have an enum for it yet?